### PR TITLE
add check for windows eol

### PR DESCRIPTION
**Pull request by darkdarkdragon**

### CODE REVIEW

**Comment by darkdarkdragon** [comment]
@clark800 done

File: scripts/checkeol.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash 
+
+function checkEol {
+  local X=$(git --no-pager diff --name-only -M100% --diff-filter=AM --relative $(git merge-base FETCH_HEAD origin/HEAD) FETCH_HEAD | grep "\.js$")
```
Comment (clark800) [inline scripts/checkeol.sh]: We should probably also check JSON files, and might as well check all file types, so we can get rid of the `grep`. Should use a better variable name than `X`.

File: scripts/checkeol.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash 
+
+function checkEol {
```
Comment (clark800) [inline scripts/checkeol.sh]: Can we call it `checkEOL`? I think it's more conventional to capitalize `EOL`.

File: scripts/checkeol.sh (hunk 1 above)
Comment (darkdarkdragon) [inline scripts/checkeol.sh]: ok

File: scripts/checkeol.sh (hunk 2 above)
Comment (darkdarkdragon) [inline scripts/checkeol.sh]: ok

File: scripts/ci.sh [hunk 3]
```diff
@@ -47,6 +51,7 @@ doctest() {
 }
 
 oneNode() {
+  checkEol
```
Comment (clark800) [inline scripts/ci.sh]: Can you change to `checkEOL` here too?

**Comment by darkdarkdragon** [comment]
done

**Comment by clark800** [comment]
LGTM

**Pull request merged**